synapse/storage/state.py

```diff
@@ -259,10 +262,17 @@ def _get_state_groups_from_groups_txn(self, txn, groups, types=None):
                         key = (typ, state_key)
                         results[group][key] = event_id
         else:
+            where_args = []
+            where_clauses = []
```

Comment:
it'd be nice to define this inside the `if` clause where it is used

Reply:
i may be going insane, but i'm failing to see why? it's used outside the if clause, so isn't it nicer to define it at the 'right' scope level, rather than give the impression that it's scoped to the if block (even though python scoping extends to the whole def?)

Reply:
*I* may be going insane, but I can't see it being used outside the `if` clause?

Reply:
surely where_args & where_clauses are used much later on to actually execute the query. hence defining them at the scope common to both. or are we talking about different if clauses?

Reply:
I'm talking about `where_clauses`, which is used at lines 270, 273 and 275. They are all inside the next if clause.

https://github.com/matrix-org/synapse/blob/b2aba9e43053f9f297671fce0051bfc18a8b655a/synapse/storage/state.py#L267-L275